src/circuit/gadget/ecc.rs

```diff
@@ -376,3 +378,168 @@ impl<C: CurveAffine, EccChip: EccInstructions<C> + Clone + Debug + Eq> FixedPoin
+        dev::MockProver,
+        pasta::pallas,
+        plonk::{Assignment, Circuit, ConstraintSystem, Error},
+    };
+
+    use super::chip::{EccChip, EccConfig, OrchardFixedBases, OrchardFixedBasesShort};
+
+    struct MyCircuit<C: CurveAffine> {
+        _marker: std::marker::PhantomData<C>,
+    }
+
+    #[allow(non_snake_case)]
+    impl<C: CurveAffine> Circuit<C::Base> for MyCircuit<C> {
+        type Config = EccConfig;
+
+        fn configure(meta: &mut ConstraintSystem<C::Base>) -> Self::Config {
+            let bits = meta.advice_column();
+            let P = (meta.advice_column(), meta.advice_column());
+            let lambda = (meta.advice_column(), meta.advice_column());
+            let extras = [
+                meta.advice_column(),
+                meta.advice_column(),
+                meta.advice_column(),
+                meta.advice_column(),
+                meta.advice_column(),
+            ];
+
+            EccChip::<C>::configure(meta, bits, P, lambda, extras)
+        }
+
+        fn synthesize(
+            &self,
+            cs: &mut impl Assignment<C::Base>,
+            config: Self::Config,
+        ) -> Result<(), Error> {
+            let mut layouter = SingleChipLayouter::new(cs)?;
+            let loaded = EccChip::<C>::load();
+            let chip = EccChip::construct(config, loaded);
+
+            // Generate a random point P
+            let p_val = C::CurveExt::random(rand::rngs::OsRng).to_affine(); // P
+            let p = super::Point::new(chip.clone(), layouter.namespace(|| "point"), Some(p_val))?;
+
+            // Generate a random point Q
+            let q_val = C::CurveExt::random(rand::rngs::OsRng).to_affine(); // P
+            let q = super::Point::new(chip.clone(), layouter.namespace(|| "point"), Some(q_val))?;
+
+            // Check complete addition point P + Q
+            {
+                let real_added = p_val + q_val;
+                let added_complete = p.add(layouter.namespace(|| "P + Q"), &q)?;
+                if let (Some(x), Some(y)) =
+                    (added_complete.inner.x.value, added_complete.inner.y.value)
+                {
+                    if C::from_xy(x, y).is_some().into() {
+                        assert_eq!(real_added.to_affine(), C::from_xy(x, y).unwrap());
+                    }
+                }
+            }
+
+            // Check incomplete addition point P + Q
+            {
+                let real_added = p_val + q_val;
+                let added_incomplete = p.add_incomplete(layouter.namespace(|| "P + Q"), &q)?;
+                if let (Some(x), Some(y)) = (
+                    added_incomplete.inner.x.value,
+                    added_incomplete.inner.y.value,
+                ) {
+                    if C::from_xy(x, y).is_some().into() {
+                        assert_eq!(real_added.to_affine(), C::from_xy(x, y).unwrap());
+                    }
+                }
+            }
```

Comment:
Check that trying to perform incomplete addition of P+P, P+(-P), P+𝒪, 𝒪+P, or 𝒪+𝒪 results in errors.

Reply:
I've modified the test to expect that these cases produce errors.